server/src/main/scala/net/wiringbits/services/UsersService.scala

```diff
@@ -83,6 +89,35 @@ class UsersService @Inject() (
     } yield Login.Response(user.id, user.name, user.email, token)
   }
 
+  def forgotPassword(request: ForgotPassword.Request): Future[ForgotPassword.Response] = {
+    for {
+      _ <- validateCaptcha(request.captcha)
+      userMaybe <- repository.find(request.email)
+      _ = userMaybe.foreach(user => {
+        if (user.verifiedOn.isEmpty)
+          throw new RuntimeException(s"User's email ${request.email} hasn't been verified yet")
+        val emailMessage = EmailMessage.forgotPassword(user.name, webAppConfig.host, s"${user.id}")
+        emailApi.sendEmail(EmailRequest(user.email, emailMessage))
+      })
+    } yield ForgotPassword.Response()
+  }
+
+  def resetPassword(userId: UUID, password: Password): Future[ResetPassword.Response] = {
+    val userMaybe = repository.find(userId)
+    userMaybe.flatMap {
+      case None => Future(ResetPassword.Response(None))
+      case Some(user) =>
+        val hashedPassword = BCrypt.hashpw(password.string, BCrypt.gensalt())
+        for {
+          _ <- repository.resetPassword(userId, hashedPassword)
+          _ <- userLogsRepository.create(user.id, "Password was reset")
```

Comment:
The logs need to be created in the transaction where the password is updated so that both operations are done atomically